core/src/test/java/org/apache/iceberg/rest/TestRESTViewCatalog.java

```diff
@@ -144,6 +151,36 @@ public void closeCatalog() throws Exception {
     }
   }
 
+  @Test
+  public void testPaginationForListViews() {
+    RESTCatalogAdapter adapter = Mockito.spy(new RESTCatalogAdapter(backendCatalog));
+    RESTCatalog catalog =
+        new RESTCatalog(SessionCatalog.SessionContext.createEmpty(), (config) -> adapter);
+    catalog.initialize("test", ImmutableMap.of(RESTSessionCatalog.REST_PAGE_SIZE, "10"));
+
+    int numberOfItems = 100;
+    String namespaceName = "newdb";
+    String tableName = "newtable";
+    String viewName = "newview";
+
+    // create initial namespace
+    catalog().createNamespace(Namespace.of(namespaceName));
+
+    // create several views under namespace, based off a table for listing and verify
+    for (int i = 0; i < numberOfItems; i++) {
+      TableIdentifier viewIndentifier = TableIdentifier.of(namespaceName, viewName + i);
+      catalog
+          .buildView(viewIndentifier)
+          .withSchema(SCHEMA)
+          .withDefaultNamespace(viewIndentifier.namespace())
+          .withQuery("spark", "select * from " + namespaceName + "." + tableName)
```

Comment:
```suggestion
          .withQuery("spark", "select * from ns.tbl")
```

and then you can remove the `tableName` variable

Reply:
will fix